src/eks-mcp-server/awslabs/eks_mcp_server/k8s_handler.py

```diff
@@ -49,16 +49,18 @@ def __init__(
         mcp,
         allow_write: bool = False,
         allow_sensitive_data_access: bool = False,
+        client_cache=None,
     ):
         """Initialize the Kubernetes handler.
 
         Args:
             mcp: The MCP server instance
             allow_write: Whether to enable write access (default: False)
             allow_sensitive_data_access: Whether to allow access to sensitive data (default: False)
+            client_cache: Optional K8sClientCache instance to share between handlers
         """
         self.mcp = mcp
-        self.client_cache = K8sClientCache()
+        self.client_cache = client_cache if client_cache is not None else K8sClientCache()
```

Comment:
Is this check needed if we are initializing K8sClientsCache in server.py to be shared? It's also not in eks_resiliency_handler.py